### PR TITLE
Mixins, CLI, documentation

### PR DESCRIPTION
**Pull request by janbrohl**
Actually, this got way bigger than expected...

- Move directive functionality to mixins so the same code can be used for CLI and Sphinx. ( #21 )
- Move documentation to code and generate from there. (#38 #24 )
- Generate CLI documentation from code. ( #3 )

### CODE REVIEW

File: doc/source/using.rst [hunk 1]
```diff
@@ -103,7 +18,16 @@ available as a global variable.
     to handle aspects like ``None`` values and fields that have values
     that may interfere with parsing reStructuredText.
 
+Directives
+==========
+
+.. datatemplate:import-module::
+      :source: sphinxcontrib.datatemplates.domain
+      
+      {% for key,directive in data.DataTemplateDomain.directives.items() %}
```
Comment (dhellmann) [inline doc/source/using.rst]: We probably want to sort them so the order is consistent and predictable.

**Review by dhellmann — COMMENTED** [review]
I hesitated over this for a little while. I think the user experience would be better if we automatically detected the data format based on the input file name, but I guess we decided with the directives that that won't always work as well as we would like.

Do you want to add an entry point for a console script before we merge this?

File: doc/source/using.rst (hunk 1 above)
Comment (janbrohl) [reply doc/source/using.rst]: Very true, sorting in 73f177e4b085099619e1f503d5fab580db258904

**Comment by janbrohl** [comment]
> 
> 
> I hesitated over this for a little while. I think the user experience would be better if we automatically detected the data format based on the input file name, but I guess we decided with the directives that that won't always work as well as we would like.
> 
> Do you want to add an entry point for a console script before we merge this?

Data format autodetection is difficult if not using file extensions but in many cases that should be enough. For this I would suggest using a new branch/pr as this one is kind of bloated already. #40

Entry points added in 97dfb1d08933386add9bce8522985073357e4d8f

**Comment by janbrohl** [comment]
Reimplementing smaller parts of this.

**Pull request closed without merging**